### DISCUSSION TITLE
Unable to insert new Conference via CLI

### DISCUSSION THREAD

**Discussion by VitaliyMarkelov** [symfony/symfony#54886] [Symfony Book: The Fast Track] [answered]
I got up to [Customizing EasyAdmin](https://symfony.com/doc/6.4/the-fast-track/en/9-backend.html#customizing-easyadmin) and everything worked fine so far. I wanted to insert a new record into conference table (PostgreSQL):
`INSERT INTO conference (city, year) VALUES ('Hamburg', 2024)`
and got an error:

> ERROR:  null value in column "id" of relation "conference" violates not-null constraint
> DETAIL:  Failing row contains (null, Hamburg, 2024, null).

I would love to get advice on why id is not incrementing itself while migration looks like this:
```
//...
$this->addSql('CREATE SEQUENCE conference_id_seq INCREMENT BY 1 MINVALUE 1 START 1');
//...
$this->addSql('CREATE TABLE conference (id INT NOT NULL, city VARCHAR(255) NOT NULL, year VARCHAR(4) NOT NULL, is_international BOOLEAN NOT NULL, PRIMARY KEY(id))');
```
and what would be the proper way to insert new records via CLI into a table with autoincrementing id. I have thoroughly googled the subject and found a suggestion:
```
SELECT MAX(id) FROM conference; 
SELECT nextVal('"conference_id_seq"');
SELECT setval('"conference_id_seq"', (SELECT MAX(id) FROM conference)+1);
```
but found nothing similar in FastTrack. How do I properly create autoincrementing id via symfony console make:entity if possible at all?

**Comment by MatTheCat** [comment]
Your migration is weird 😅 

Please provide the output of `composer show doctrine/dbal`, and `Conference::$id` declaration including all of Doctrine's annotations/attributes.

**Comment by VitaliyMarkelov** [comment]
> name     : doctrine/dbal
> descrip. : Powerful PHP database abstraction layer (DBAL) with many features for database schema introspection and management.
> keywords : abstraction, database, db2, dbal, mariadb, mssql, mysql, oci8, oracle, pdo, pgsql, postgresql, queryobject, sasql, sql, sqlite, sqlserver, sqlsrv
> versions : * 3.8.4
> type     : library
> license  : MIT License (MIT) (OSI approved) https://spdx.org/licenses/MIT.html#licenseText
> homepage : https://www.doctrine-project.org/projects/dbal.html
> source   : [git] https://github.com/doctrine/dbal.git b05e48a745f722801f55408d0dbd8003b403dbbd
> dist     : [zip] https://api.github.com/repos/doctrine/dbal/zipball/b05e48a745f722801f55408d0dbd8003b403dbbd b05e48a745f722801f55408d0dbd8003b403dbbd
> path     : C:\PROJECTS\guestbook\vendor\doctrine\dbal
> names    : doctrine/dbal
> 
> support
> issues : https://github.com/doctrine/dbal/issues
> source : https://github.com/doctrine/dbal/tree/3.8.4
> 
> autoload
> psr-4
> Doctrine\DBAL\ => src
> 
> requires
> composer-runtime-api ^2
> doctrine/cache ^1.11|^2.0
> doctrine/deprecations ^0.5.3|^1
> doctrine/event-manager ^1|^2
> php ^7.4 || ^8.0
> psr/cache ^1|^2|^3
> psr/log ^1|^2|^3
> 
> requires (dev)
> doctrine/coding-standard 12.0.0
> fig/log-test ^1
> jetbrains/phpstorm-stubs 2023.1
> phpstan/phpstan 1.10.58
> phpstan/phpstan-strict-rules ^1.5
> phpunit/phpunit 9.6.16
> psalm/plugin-phpunit 0.18.4
> slevomat/coding-standard 8.13.1
> squizlabs/php_codesniffer 3.9.0
> symfony/cache ^5.4|^6.0|^7.0
> symfony/console ^4.4|^5.4|^6.0|^7.0
> vimeo/psalm 4.30.0

```
namespace App\Entity;

use App\Repository\ConferenceRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ConferenceRepository::class)]
class Conference
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    private ?string $city = null;

    #[ORM\Column(length: 4)]
    private ?string $year = null;

    #[ORM\Column]
    private ?bool $isInternational = null;

    /**
     * @var Collection<int, Comment>
     */
    #[ORM\OneToMany(targetEntity: Comment::class, mappedBy: 'conference', orphanRemoval: true)]
    private Collection $comments;

    public function __construct()
    {
        $this->comments = new ArrayCollection();
    }


	public function __toString(): string
    {
        return $this->city.' '.$this->year;
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getCity(): ?string
    {
        return $this->city;
    }

    public function setCity(string $city): static
    {
        $this->city = $city;

        return $this;
    }

    public function getYear(): ?string
    {
        return $this->year;
    }

    public function setYear(string $year): static
    {
        $this->year = $year;

        return $this;
    }

    public function isInternational(): ?bool
    {
        return $this->isInternational;
    }

    public function setInternational(bool $isInternational): static
    {
        $this->isInternational = $isInternational;

        return $this;
    }

    /**
     * @return Collection<int, Comment>
     */
    public function getComments(): Collection
    {
        return $this->comments;
    }

    public function addComment(Comment $comment): static
    {
        if (!$this->comments->contains($comment)) {
            $this->comments->add($comment);
            $comment->setConference($this);
        }

        return $this;
    }

    public function removeComment(Comment $comment): static
    {
        if ($this->comments->removeElement($comment)) {
            // set the owning side to null (unless already changed)
            if ($comment->getConference() === $this) {
                $comment->setConference(null);
            }
        }

        return $this;
    }
}
```

Yeah, probably my thinking is weird as I never touched SF before ). I've checked id definitions and compared them with the file I downloaded from the [repo](https://github.com/the-fast-track/book-6.2-1), they look the same, at least here:
`    #[ORM\GeneratedValue]`

What I did I changed id to this:
`    #[ORM\GeneratedValue(strategy: "IDENTITY")]`
and this solved my problem. But as I am learning I would like to get down to the core of the problem, where I did a mistake.

**Reply by MatTheCat** [reply, marked as answer]
Ok so I think your original issue is that the sequence created by Doctrine is not used as `DEFAULT` for your ID, which would match https://github.com/doctrine/dbal/issues/3619#issuecomment-511619233.

Still, you should configure your strategy as `SEQUENCE`, because the PostgreSQL `IDENTITY` strategy will change when you upgrade DBAL to v4 (see https://github.com/doctrine/orm/blob/4.0.x/UPGRADE.md#deprecated-reliance-on-the-non-optimal-defaults-that-come-with-the-auto-identifier-generation-strategy)

So, you’d have to get the sequence’s next value by yourself (using `SELECT NEXTVAL('conference_id_seq')`) and assign it to the ID.

But in fact the ORM would do it for you if you used it. Why do you want to insert rows by hand?

**Reply by VitaliyMarkelov** [reply]
> Why do you want to insert rows by hand?
I am from punchcard era, can't change ).
